barretenberg/cpp/src/barretenberg/honk/transcript/transcript.hpp

```diff
@@ -85,17 +85,17 @@ template <typename FF> class BaseTranscript {
      */
     [[nodiscard]] std::array<uint8_t, HASH_OUTPUT_SIZE> get_next_challenge_buffer() const
     {
+        // Empty buffer to compare against
+        std::array<uint8_t, HASH_OUTPUT_SIZE> empty_challenge_buffer{};
         // Prevent challenge generation if nothing was sent by the prover.
-        ASSERT(!current_round_data.empty());
+        ASSERT(!(current_round_data.empty() && previous_challenge_buffer == empty_challenge_buffer));
 
         // concatenate the hash of the previous round (if not the first round) with the current round data.
```

Comment:
Update Adrian's comment (here and elsewhere) to ensure it's up to date with your changes